src/libstd/sys/windows/os.rs

```diff
@@ -78,22 +78,28 @@ impl Iterator for Env {
     type Item = (OsString, OsString);
 
     fn next(&mut self) -> Option<(OsString, OsString)> {
-        unsafe {
-            if *self.cur == 0 { return None }
-            let p = &*self.cur;
-            let mut len = 0;
-            while *(p as *const u16).offset(len) != 0 {
-                len += 1;
+        loop {
+            unsafe {
+                if *self.cur == 0 { return None }
+                let p = &*self.cur as *const u16;
+                let mut len = 0;
+                while *p.offset(len) != 0 {
+                    len += 1;
+                }
+                let s = slice::from_raw_parts(p, len as usize);
+                self.cur = self.cur.offset(len + 1);
+
+                // `s` has at least length 1 at this point, because the empty
+                // string terminates the array of environment variables.
```

Comment:
Can you update this comment to mention that the purpose of this slice-after-one is not because we know it has length one but because we want to skip any leading `=` character?